langwatch/src/optimization_studio/server/loadDatasets.ts

```diff
@@ -73,28 +73,19 @@ export const loadDatasets = async (
         };
       }
 
-      // Select from database dataset
+      // For database datasets, we'll just pass the ID and let the Python SDK fetch it
       if (!node.data.dataset.id) {
         throw new Error("Dataset ID is required");
       }
 
-      const dataset = await getFullDataset({
-        datasetId: node.data.dataset.id,
-        projectId,
-        entrySelection,
-      });
-      if (!dataset) {
-        throw new Error("Dataset not found");
-      }
-      const inMemoryDataset = datasetDatabaseRecordsToInMemoryDataset(dataset);
-      delete inMemoryDataset.datasetId;
-      const inlineDataset = inMemoryDatasetToNodeDataset(inMemoryDataset);
```

Comment:
one problem I see of moving this entirely to the python backend now, is that the way you did it:

- if I select only 1 entry, on a dataset of 25mb, still I'm downloading the 25mb every time, to extract only that single index

can we keep the inlineDataset here IIF the entrySelection is not "all" (first, last or index)?

this also makes it much easier to debug when we run a single entry, we can see which date is going down the wirte, keep the backend fetching for when we really need the full dataset, now the backend knows how to handle if inline dataset is not available